components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -312,12 +317,33 @@ private void addEndpoint(ActionDTO actionDTO, Integer tenantId) throws ActionMgt
             endpoint.getAuthentication().getProperties().forEach(
                     authProperty -> endpointProperties.put(authProperty.getName(),
                             new ActionProperty.BuilderForDAO(authProperty.getValue()).build()));
+            endpointProperties.put(ALLOWED_HEADERS_PROPERTY, createActionProperty(endpoint.getAllowedHeaders()));
+            endpointProperties.put(ALLOWED_PARAMETERS_PROPERTY, createActionProperty(endpoint.getAllowedParameters()));
             addActionPropertiesToDB(actionDTO.getId(), endpointProperties, tenantId);
         } catch (TransactionException e) {
             throw new ActionMgtServerException("Error while adding Action Endpoint configurations in the system.", e);
         }
     }
 
+    /**
+     * Creates an ActionProperty object from a list of attributes by converting them into a JSON string.
+     *
+     * @param attributes List of attributes to be converted into an ActionProperty.
+     * @return An ActionProperty object containing the JSON representation of the attributes.
+     * @throws ActionMgtServerException If an error occurs while converting the attributes to a JSON string.
+     */
+    private ActionProperty createActionProperty(List<String> attributes) throws ActionMgtServerException {
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            BinaryObject attributesBinaryObject = BinaryObject.fromJsonString(objectMapper
+                    .writeValueAsString(attributes));
+            return new ActionProperty.BuilderForDAO(attributesBinaryObject).build();
+        } catch (JsonProcessingException e) {
+            throw new ActionMgtServerException("Failed to convert object values to JSON string.", e);
+        }
+    }
```

Comment:
Better move to a util if there are data layer utils defined already or make it clear to mentioning that this converts a string property list to an action property object